client/src/controllers/InitController.test.js

```diff
@@ -148,4 +148,39 @@ describe('InitController', () => {
       ]);
     });
   });
+
+  describe('with a detail', () => {
+    const handleEvent = jest.fn();
+    document.addEventListener('w-init:ready', handleEvent);
+
+    beforeAll(() => {
+      jest.clearAllMocks();
+
+      application?.stop();
+
+      document.body.innerHTML = `
+        <div
+          id="test"
+          class="test-detail"
+          data-controller="w-init"
+          data-w-init-detail-value='{"someMessage": "some value"}'
+          Test body
+        </div>
+      `;
```

Comment:
The HTML as it's written is malformed, the closing brace on the div opening tag is missing.

Also we should test that any custom events dispatches also get the detail event.

Here's a refined suggestion.

```suggestion
      document.body.innerHTML = `
        <article
          id="test"
          class="test-detail"
          data-controller="w-init"
          data-w-init-detail-value='${JSON.stringify(detail)}'
          data-w-init-event-value='my-custom:event'
        >
          Test body
        </article>
      `;
```

Reply:
Thanks a bunch for this part. When I first wrote this test, I actually used a `detail` variable, but when I passed it to the HTML, it kept crashing. 😅 
This resolves this nicely.